pkg/broker/user_info.go

```diff
@@ -0,0 +1,62 @@
+package broker
+
+import (
+	"encoding/json"
+	"fmt"
+
+	osb "github.com/pmorie/go-open-service-broker-client/v2"
+)
+
+// ParseKubernetesIdentity - creates a kubernetes identity from the
+// orginating identity
+func ParseKubernetesIdentity(o osb.OriginatingIdentity) (*K8SIdentity, error) {
+	if o.Platform != "kubernetes" {
+		return nil, fmt.Errorf("invalid plateform value, expecting kubernetes")
+	}
+
+	u := UserInfo{}
+	err := json.Unmarshal([]byte(o.Value), &u)
```

Comment:
I noticed in the `ParseCloudFoundryIdentity` we validate if the `user_id` is present or not. Are there similar requirements for when the platform is K8s?

Reply:
This is what I was attempting to work off of: https://github.com/openservicebrokerapi/servicebroker/blob/master/profile.md#kubernetes-originating-identity-header

I initially thought that the `json.Unmarshal` would make sure that this is the correct object. 

@pmorie @LiliC should we add `nil` checks for `Extras` and `Groups` and make `Username` and `UID` type defined pointers to a string to test if they are `nil` or not as well?